src/WidgetBase.ts

```diff
@@ -284,7 +285,14 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 	}
 
 	private _setNode(element: Element, properties: VNodeProperties): void {
-		this._nodeMap.set(String(properties.key), <HTMLElement> element);
+		const key = String(properties.key);
+		this._nodeMap.set(key, <HTMLElement> element);
+		const callbacks = this._requiredNodes.get(key);
+		if (callbacks) {
+			for (const callback of callbacks) {
+				callback.call(this, element);
```

Comment:
does this not now then bleed the scope of the widget into a meta callback?

Reply:
Yes. Will fix. Don't need to pass element either.

Reply:
👍